docs/guides/upgrading-v3.md

```diff
@@ -0,0 +1,265 @@
+
+  This is the simplest. Upgrade the global version to v3, and install v2 in specific projects (via NPM). The `serverless` command will automatically run the correct version (v3 can run v2).
+
+- Using v2 globally, and v3 in specific projects.
+
+  To achieve that, install v3 in specific projects (via NPM). Then, use `serverless` for v2 projects, and `npx serverless` for v3 projects.
+
+## Breaking changes
+
+You will find below a complete list of all breaking changes. All those breaking changes were signaled via deprecation messages in Serverless Framework v2.
+
+### CLI commands and options
+
+The `serverless` CLI no longer runs on Node v10 because [that version is obsolete](https://endoflife.date/nodejs): upgrade to v12 or greater to run `serverless` on your machine.
+
+The `serverless` CLI used to accept free-form CLI options. This feature was deprecated and has been removed. The main reason is that this prevented us from detecting typos in options, which sometimes created unexpected situations and overall a bad user experience. [Learn more about this change](../deprecations.md#handling-of-unrecognized-cli-options).
+
+Additionally, all CLI options must now be passed at the end of the commands:
+
+```bash
+# Will no longer work in v3:
+serverless --verbose deploy
+
+# Correct syntax:
+serverless deploy --verbose
+```
+
+This change makes the CLI much more robust at detecting arguments from options and their values.
+
+On that note, the `-v` option is no longer recognized (it was ambiguous with `--version`): use the full `--verbose` option instead.
+
+When the `serverless` CLI is installed globally and locally (in the project’s `node_modules`), the local version will always be used. It is no longer possible to disable that behavior ([learn more](../deprecations.md#support-for-enablelocalinstallationfallback-setting-is-to-be-removed)).
+
+Finally, the `serverless studio` command has been removed: that feature was deprecated and is no longer available.
+
+### Service configuration
+
+The default Lambda runtime has changed from NodeJS 12 to NodeJS 14, given this is now the default runtime recommended by AWS.
+
+In `serverless.yml`, the `service` key no longer accepts a YAML object ([learn more](../deprecations.md#service-property-object-notation)).
+
+```yaml
+# Deprecated in v2, removed in v3:
+service:
+    name: my-service
+
+# Correct syntax:
+service: my-service
+```
+
+All options that used to be defined inside the `service` key have been moved to other sections (mentioned below in that document). This change clears up confusion that existed between the `service` and `provider` sections.
+
+CloudFormation extensions defined in `resources.extensions` are now validated: extensions that point to unknown resources will throw an error, which should help catch broken configuration before deploying ([learn more](../deprecations.md#defining-extensions-to-nonexistent-resources-in-resourcesextensions)).
```

Comment:
This paragraph can be removed, as this change was made in context of v2 (it was logical error to configure it as deprecation)